data-model/src/main/java/io/micronaut/data/model/query/builder/AbstractSqlLikeQueryBuilder.java

```diff
@@ -638,7 +642,7 @@ private void buildSelectClause(QueryModel query, QueryState queryState, StringBu
      * @return True if they should be escaped
      */
     protected boolean shouldEscape(@NonNull PersistentEntity entity) {
-        return entity.getAnnotationMetadata().booleanValue(MappedEntity.class, "escape").orElse(true);
+        return entity.getAnnotationMetadata().booleanValue(MappedEntity.class, "escape").orElse(true) && !entity.isJsonView();
```

Comment:
Because it turns field into uppercase json fields, so `usrView.DATA.usrId` would become `usrView.DATA."USRID "` and field would not be resolvable in the result. But, I see now it's not the best place, it should be moved to use field name vs persisted name which this would not solve.

Reply:
Yeah probably, escape is just for SQL columns

Reply:
Changed this.